packages/next/src/client/components/unstable-rethrow.ts

```diff
@@ -0,0 +1,26 @@
+import { isDynamicUsageError } from '../../export/helpers/is-dynamic-usage-error'
+import { isPostpone } from '../../server/lib/router-utils/is-postpone'
+import { isBailoutToCSRError } from '../../shared/lib/lazy-dynamic/bailout-to-csr'
+import { isNextRouterError } from './is-next-router-error'
+
+/**
+ * This function should be used to rethrow internal Next.js errors so that they can be handled by the framework.
+ * When wrapping an API that uses errors to interrupt control flow, you should use this function before you do any error handling.
+ * This function will rethrow the error if it is a Next.js error so it can be handled, otherwise it will do nothing.
+ *
+ * Read more: [Next.js Docs: `unstable_rethrow`](https://nextjs.org/docs/app/api-reference/functions/unstable_rethrow)
+ */
+export function unstable_rethrow(error: unknown): void | never {
```

Comment:
The `never` is unnecessary here and does nothing

```suggestion
export function unstable_rethrow(error: unknown): void {
```

`_ | never` is a noop, i.e. `X | never` is `X`. See [TS playground](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXzQGcMoAjCEAfTgwAsYcB3AChBgZgC4DUBrVJqgCU3AG44sweAB94qEKLYAoJWDzF4AD3gBeHsTIVqIOgxYBmIUs1KA9LfgA9APwqgA)

...which kinda makes sense -- it's just a void function that throws *sometimes*.

Reply:
You're right -- copied this from the [RFC](https://github.com/vercel/next.js/discussions/64076) signature 🙈